internal/linker/link_node_modules.ts

```diff
@@ -0,0 +1,225 @@
+/**
+ * @fileoverview Creates a node_modules directory in the current working directory
+ * and symlinks in the node modules needed to run a program.
+ * This replaces the need for custom module resolution logic inside the process.
+ */
+import * as fs from 'fs';
+import * as path from 'path';
+
+// Run Bazel with --define=VERBOSE_LOGS=1 to enable this logging
+const VERBOSE_LOGS = !!process.env['VERBOSE_LOGS'];
+
+function log_verbose(...m: string[]) {
+  if (VERBOSE_LOGS) console.error('[link_node_modules.js]', ...m);
+}
+
+function panic(m: string) {
+  throw new Error(`Internal error! Please run again with
+   --define=VERBOSE_LOG=1
+and file an issue: https://github.com/bazelbuild/rules_nodejs/issues/new?template=bug_report.md
+Include as much of the build output as you can without disclosing anything confidential.
+
+  Error:
+  ${m}
+  `);
+}
+
+function symlink(target: string, path: string) {
+  if (fs.existsSync(path)) {
+    // We assume here that the path is already linked to the correct target.
+    // Could add some logic that asserts it here, but we want to avoid an extra
+    // filesystem access so we should only do it under some kind of strict mode.
+    return;
+  }
+  log_verbose(`symlink( ${path} -> ${target} )`);
+  // Use junction on Windows since symlinks require elevated permissions.
+  // We only link to directories so junctions work for us.
+  fs.symlinkSync(target, path, 'junction');
+  if (VERBOSE_LOGS) {
```

Comment:
maybe `if (VERBOSE_LOGS || DEBUG)` so if you set DEBUG env you're checking for this? Maybe throw in that case as well since seems like failure you should find out about if you set DEBUG?

Reply:
I didn't want to throw because I think it's probably wrong for the program to have different behavior when you turn on VERBOSE_LOGS - seems like we want a strict mode in general which is maybe different from how much logging we do

Reply:
I was thinking just throw if DEBUG was set not on VERBOSE_LOGS. Program can have different behaviors with DEBUG.

Reply:
But maybe a strict mode would make more sense instead of overloading the DEBUG semantics